drive.go

```
@@ -0,0 +1,158 @@
+package deta
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+)
+
+const (
+	uploadChunkSize = 1024 * 1024 * 10
+)
+
+var (
+	// ErrEmptyName empty name 
+	ErrEmptyName    = errors.New("name is empty")
+	// ErrEmptyNames empty names 
+	ErrEmptyNames   = errors.New("names is empty")
+	// ErrTooManyNames too many items 
+	ErrTooManyNames = errors.New("more than 1000 files to delete")
+	// ErrEmptyData no data
+	ErrEmptyData = errors.New("no data provided")
+)
+
+// Drive is a Deta Drive service client that offers the API to make requests to Deta Drive
+type Drive struct {
+	// deta api client
+	client *detaClient
+
+	// auth info for authenticating requests
+	auth *authInfo
+}
+
+// NewDrive returns a pointer to new Drive
+func newDrive(projectKey, driveName, rootEndpoint string) *Drive {
+	parts := strings.Split(projectKey, "_")
+	projectID := parts[0]
+
+	// root endpoint for the base
+	rootEndpoint = fmt.Sprintf("%s/%s/%s", rootEndpoint, projectID, driveName)
+
+	return &Drive{
+		client: newDetaClient(rootEndpoint, &authInfo{
+			authType:    "api-key",
+			headerKey:   "X-API-Key",
+			headerValue: projectKey,
+		}),
+	}
+}
+
+type startUploadResponse struct {
+	UploadID  string `json:"upload_id"`
+	Name      string `json:"name"`
+	ProjectID string `json:"project_id"`
+	DriveName string `json:"drive_name"`
+}
+
+// Initializes a chuncked file upload.
+// 
+// If successful, returns the UploadID
+func (d *Drive) startUpload(name string) (string, error) {
+	url := fmt.Sprintf("/uploads?name=%s", name)
+	o, err := d.client.request(&requestInput{
+		Path: url,
+		Method: "POST",
+	})
+	if err != nil {
+		return "", err
+	}
+
+	var sr startUploadResponse 
+	err = json.Unmarshal(o.Body, &sr)
+	if err != nil {
+		return "", err
+	}
+
+	return sr.UploadID, nil
+}
+
+// End a chuncked upload.
+func (d *Drive) finishUpload(name, uploadId string) {
```

should return an error and we should not ignore errors